$OpenBSD: patch-lib_rubygems_ext_ext_conf_builder_rb,v 1.1.1.1 2017/01/06 15:59:25 jeremy Exp $

Ugly hack to make --user-install option work.  Without this, when
a user uses gem install --user-install, it calls
/usr/bin/install -o root -g bin, which fails due to permission issues.
This removes the -o root -g bin, so it can succeed as a regular user.

Don't output mkmf.log error message if compilation didn't fail.

--- lib/rubygems/ext/ext_conf_builder.rb.orig	Mon Feb  1 04:43:26 2016
+++ lib/rubygems/ext/ext_conf_builder.rb	Thu Dec 29 10:47:56 2016
@@ -36,6 +36,11 @@ class Gem::Ext::ExtConfBuilder < Gem::Ext::Builder
         siteconf.puts "RbConfig::MAKEFILE_CONFIG['#{dir}'] = dest_path"
         siteconf.puts "RbConfig::CONFIG['#{dir}'] = dest_path"
       end
+      unless Process.euid == 0
+        %w[INSTALL INSTALL_DATA INSTALL_SCRIPT INSTALL_PROGRAM].each do |ins|
+          siteconf.puts "RbConfig::MAKEFILE_CONFIG['#{ins}'] = RbConfig::CONFIG['#{ins}'] = RbConfig::CONFIG['#{ins}'].gsub(/-o root -g bin/, '')"
+        end
+      end
 
       siteconf.flush
 
@@ -48,9 +53,11 @@ class Gem::Ext::ExtConfBuilder < Gem::Ext::Builder
           run cmd, results
         ensure
           if File.exist? 'mkmf.log'
-            results << "To see why this extension failed to compile, please check" \
-              " the mkmf.log which can be found here:\n"
-            results << "  " + File.join(dest_path, 'mkmf.log') + "\n"
+            unless $?.success? then
+              results << "To see why this extension failed to compile, please check" \
+                " the mkmf.log which can be found here:\n"
+              results << "  " + File.join(dest_path, 'mkmf.log') + "\n"
+            end
             FileUtils.mv 'mkmf.log', dest_path
           end
           siteconf.unlink
